### PR TITLE
Empty Stats: Fix presentation logic

### PR DESCRIPTION
**Pull request by nikola-milicevic**
Fixes [#17308](https://github.com/wordpress-mobile/WordPress-iOS/issues/17308)

This PR fixes the presentation logic.

## To test:
1. Delete the app and reinstall it (to clear the cache)
2. Go to My **Site** > **Stats** for site with less than 30 views
3. Complete all the Nudges (or dismiss them)
4. Return to **Site** > **Stats** screen and complete the Customize card (or dismiss it)
5. Nudges should be shown for other sites with less than 30 views
6. Customize card should not be shown for any site

## Regression Notes
1. Potential unintended areas of impact
n/a

2. What I did to test those areas of impact (or what existing automated tests I relied on)
n/a

3. What automated tests I added (or what prevented me from doing so)
n/a

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Stats/Insights/SiteStatsPinnedItemStore.swift [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        InsightType.customize
+    ]
+    private let siteId: NSNumber
+
+    init(siteId: NSNumber) {
+        self.siteId = siteId
+    }
+
+    func itemToDisplay(for siteViewsCount: Int) -> SiteStatsPinnable? {
+        let threshold = 30
```
Comment (leandroalonso) [inline WordPress/Classes/ViewRelated/Stats/Insights/SiteStatsPinnedItemStore.swift]: I liked the `isSiteViewsCountLow` :(

Anyway, this looks way better. I'm just wondering if we should move the threshold to a class level?

**Review by leandroalonso — APPROVED** [review]
:shipit:

File: WordPress/Classes/ViewRelated/Stats/Insights/SiteStatsPinnedItemStore.swift [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        return shouldShow(item) ? item : nil
+    }
+
+    func userDefaultsKey(for item: SiteStatsPinnable) -> String {
+        switch item {
+        case is GrowAudienceCell.HintType:
+            let item = item as! GrowAudienceCell.HintType
+            return "StatsInsights-\(siteId.intValue)-\(item.rawValue)-isHidden"
+        case InsightType.customize:
+            return "StatsInsightsHideCustomizeCard"
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Stats/Insights/SiteStatsPinnedItemStore.swift:58]: This is great! Thank you for cleaning things up. What do you think of adding a comment here to provide more context?
```
// Store 'customize' separately as it is not per site.
```

**Review by momo-ozawa — APPROVED** [review]
✨ 💯 🚀 
LGTM! Works as described, left one small comment.

File: WordPress/Classes/ViewRelated/Stats/Insights/SiteStatsPinnedItemStore.swift (hunk 1 above)
Comment (nikola-milicevic) [reply WordPress/Classes/ViewRelated/Stats/Insights/SiteStatsPinnedItemStore.swift]: Thanks, I agree. Moving it up 😊

File: WordPress/Classes/ViewRelated/Stats/Insights/SiteStatsPinnedItemStore.swift (hunk 2 above)
Comment (nikola-milicevic) [reply WordPress/Classes/ViewRelated/Stats/Insights/SiteStatsPinnedItemStore.swift:58]: Thanks! Sure thing!

**Comment by nikola-milicevic** [comment]
@momo-ozawa @leandroalonso Thanks for the suggestions! I updated the code, plus did some files separation – moved `InsightType` and `SiteStatsInsightsDelegate` to its own files.

**Pull request merged**